docs/index.md

```diff
@@ -13,13 +13,24 @@ description: |-
 ## Example Usage
 
 ```terraform
+# With Dagster Cloud CLI env vars
```

Comment:
I would make it explicit that env vars don't override configuration, but are only used when the provider definition is emtpy